lib/runtime-core/src/memory/ptr.rs

```diff
@@ -137,6 +137,17 @@ impl<T: Copy + ValueType> WasmPtr<T, Array> {
         let slice: &[u8] = unsafe { std::slice::from_raw_parts(ptr, str_len as usize) };
         std::str::from_utf8(slice).ok()
     }
+
+    /// Get a UTF-8 string representation of this `WasmPtr`, where the string is nul-terminated.
+    /// Note that this does not account for UTF-8 strings that _contain_ nul themselves,
+    /// [`get_utf8_string`] has to be used for those.
+    pub fn get_utf8_string_nul_terminated<'a>(self, memory: &'a Memory) -> Option<&'a str> {
+        memory.view::<u8>()[(self.offset as usize)..]
+            .iter()
+            .map(|cell| cell.get())
+            .position(|byte| byte == 0)
+            .and_then(|length| self.get_utf8_string(memory, length.try_into().unwrap()))
```

Comment:
⛳️ 
`length.try_into().unwrap()` -> `length as u32`

`try_into` is better for being super generic but Wasmer is written in a way that there's no chance that it will work on systems with pointers smaller than 32 bits, so it's safe to just do the cast directly!  If we want to run on 16bit machines in the future, we'll have to do a ton of refactoring anyways!